contrib/pyln-client/pyln/client/plugin.py

```diff
@@ -431,13 +431,15 @@ def add_hook(self, name: str, func: Callable[..., JSONType],
 
         method = Method(name, func, MethodType.HOOK)
         method.background = background
-        method.before = before
-        method.after = after
+        if before:
+            method.before = before
+        if after:
+            method.after = after
```

Comment:
We should always set these, otherwise we'd have to check with `hasattr` whether the hook has that field.

```
method.before = [] if before else before
method.after = [] if after else after
```

Reply:
Hmm, mypy didn't like that, so went for an open-coded variant.